src/vs/workbench/contrib/search/browser/quickTextSearch/textSearchQuickAccess.ts

```diff
@@ -116,10 +122,15 @@ export class TextSearchQuickAccess extends PickerQuickAccessProvider<ITextSearch
 			if (item?.match) {
 				// we must remember our curret view state to be able to restore (will automatically track if there is already stored state)
 				this.editorViewState.set();
-				// open it
-				this._editorService.openEditor({
-					resource: item.match.parent().resource,
-					options: { preserveFocus: true, revealIfOpened: true, ignoreError: true, selection: item.match.range() }
+				const itemMatch = item.match;
+				this.editorSequencer.queue(async () => {
+					// disable and re-enable history service so that we can ignore this history entry
+					this._historyService.shouldIgnoreActiveEditorChange = true;
+					await this._editorService.openEditor({
```

Comment:
Opening an editor can fail with an error thrown and so this would leave the history in ignore-mode, without ever setting it back. At least a `try/finally` is needed.